crates/uv-resolver/src/pubgrub/priority.rs

```diff
@@ -140,6 +140,10 @@ pub(crate) enum PubGrubPriority {
     Singleton(Reverse<usize>),
 
     /// The package was specified via a direct URL.
+    ///
+    /// N.B.: URLs need to have priority over registry distributions for correctly matching registry
+    /// distributions to URLs, see [`PubGrubPackage::from_package`] an
+    /// [`crate::fork_urls::ForkUrls`].
```

Comment:
Why does this work for cases in which the URL dependency is deeper in the tree than the registry version?

Reply:
E.g., what if we fork, and add `flask == 3.0.0`, but then some other package in the forked branch depends on `flask @ https://...`. I might've imagined that we instead populate `ForkUrls` from `Urls` by filtering with the fork markers, but it looks like we instead add them by looking at the added dependencies.

Reply:
Correct me if i'm wrong, but my understanding was that we only allow url dependencies to come from urls dependencies, and we do any url dependency before any registry dependency, so if we have:
```
flask==3.0.0
foo @ https://foo.zip
```
foo:
```
flask @ https://flask.zip
``
then the order we go is `foo @ https://foo.zip`, `flask @ https://flask.zip` and `flask==3.0.0`

Reply:
But in that case, when we add the dependencies for whatever package contains `flask` and `foo` (the first package in your example), wouldn't we add a dependency on `flask` _without_ the URL component, since we haven't visited it yet?

Reply:
I removed the url field from `PubGrubPackage` to support this.